### PR TITLE
Feature/jwt Closes #45, closes #46

### LINKED ISSUE #45: Implementação JWT
Um usuário do sistema deve possuir um token para realizar requisições às rotas protegidas.

**O token deve obedecer as seguintes restrições:** 
- O token deve ser retornado no response do login
- O token deve ser válido por no máximo 30 minutos
- O token deve ser armazenado no banco de dados

- [ ] Gerar um token no login e armazená-lo no banco de dados
- [ ] Implementar testes unitários

### LINKED ISSUE #46: Middleware para restringir o tipo de usuário
Uma pessoa com conta de usuário não pode acessar as rotas de moderador. 

- [ ] Realização de testes

### PR DESCRIPTION
**Pull request by viniferraria**
# PR Description

#### Proposed changes

*Describe your changes*

#### Kind of Change
What kind of change does your code introduce? Put an ```x``` in the box that applies.
**NOTE: Please do not create MRs with different purposes. Each MR must have a single purpose: Hotfix, New Feature, or Structural Change.**

- [ ] Hotfix.    
- [X] New Feature. 
- [ ] Structural Change.  

#### Checklist

Put an ```x``` in the boxes that apply.

### This MR follows the base conventions of the project:
- [ ] Does not add commented code.
- [X] Does not add code with ```print```.
- [ ] Documentation have been added / updated as needed.
- [ ] Tests have been added / updated as needed.

### Tests:
- [X] Tests are running at 70% coverage, with success and error cases for new features, bugs, and hotfixes.
- [ ] Does not apply to this MR

### Accessibility:
- [ ] Accessibility have been implemented or updated.
- [ ] Does not apply to this MR

### Third-Party Libraries:
- [ ] A new library? It's really neccessary? Explain why
- [ ] Does not apply to this MR

### Comments:

*Comment here if necessary*

# Final checks:
- [X] **All** tests have been executed and are succeeding.

### CODE REVIEW

File: __tests__/integration/TesteAgendamento.js [hunk 1]
```diff
@@ -7,95 +7,101 @@ module.exports = () => {
...
     });
 
-    test('Deve retornar falha na falta de dados para cancelar um agendamento', async () =>{
-        const response = await request(app)
-        .delete('/agendamentos')
-        .send(AgendamentoTesteFalha);
+    // test('Deve retornar falha na falta de dados para cancelar um agendamento', async () =>{
+    //     const response = await request(app)
+    //     .delete('/agendamentos')
+    //     .set('Authorization', `Bearer ${UsuarioTeste.jwt}`)
```
Comment (zewuchen) [inline __tests__/integration/TesteAgendamento.js:45]: Remover comentários de código

File: src/app/controllers/AgendamentoController.js [hunk 2]
```diff
@@ -7,10 +7,8 @@ const { Op } = Sequelize;
 
 module.exports = {
     async getMyAgendamentos(req, res) {
-        let { user_id } = req.query;
-
         try {
-
+            let { userId } = req;
```
Comment (zewuchen) [inline src/app/controllers/AgendamentoController.js:11]: Aqui você faz de uma forma

File: src/app/controllers/BarbeariaController.js [hunk 3]
```diff
@@ -30,10 +30,9 @@ module.exports = {
 
     async getMyBarbearias(req, res) {
         try {
-            const { user_id } = req.params;
             const barbearia = await Barbearia.findOne({
                 where: { 
-                    user_id: user_id
+                    user_id: req.userId
```
Comment (zewuchen) [inline src/app/controllers/BarbeariaController.js:35]: Aqui você faz de outra

**Review by zewuchen — COMMENTED** [review]
Padronizar requisição do id do usuário no request

**Review by zewuchen — APPROVED** [review]

**Review by zewuchen — APPROVED** [review]

**Comment by viniferraria** [comment]
Closes #45 e closes #46

**Pull request merged**